vars/infra.groovy

```diff
@@ -29,29 +29,58 @@ Object checkout(String repo = null) {
     }
 }
 
+/**
+ * Retrieves Settings file to be used with Maven.
+ * If {@code MAVEN_SETTINGS_FILE_ID} variable is defined, the file will be retrieved from the specified
+ * configuration ID provided by Config File Provider Plugin.
+ * Otherwise it will fallback to a standard Jenkins infra resolution logic.
+ * @param settingsXml Absolute path to the destination settings file
+ * @param jdk Version of JDK to be used
+ * @return {@code true} if the file has been defined
+ */
+boolean retrieveMavenSettingsFile(String settingsXml, String jdk = 8) {
```

Comment:
Is `jdk` supposed to be an integer or a String? It's weird to me that it's typed as a String when only numeric operations are done

Reply:
Maybe not-as-String at some point. e.g. 8u161 or 10.021-rc